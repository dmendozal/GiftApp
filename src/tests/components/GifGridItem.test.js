import {shallow} from "enzyme";
import {GifGridItem} from "../../components/GifGridItem";

describe('Prueba del componente GifGridItem', () => {
    const title = 'Este es un titulo';
    const url = 'esto es una url';
    const animateFadeIn = 'animate__fadeInUp';
    const wrapper = shallow(<GifGridItem title={title} url={url}/>);

    test('debe mostrar el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('debe de tener un parrafo con el title', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    })

    test('debe tener la imagen igual al url y alt de los props', () => {
        const img = wrapper.find('img');
        expect(img.props().src).toBe(url);
        expect(img.props().alt).toBe(title);
    });

    test('debe de tener animate__fadeIn', () => {
        const div = wrapper.find('div');
        console.log(div.props().className);
        expect(div.prop('className').includes(animateFadeIn)).toBe(true);
    })
})