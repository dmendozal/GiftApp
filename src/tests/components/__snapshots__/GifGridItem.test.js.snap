// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba del componente GifGridItem debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__fadeInUp"
>
  <img
    alt="Este es un titulo"
    src="esto es una url"
  />
  <p
    className="title"
  >
     
    Este es un titulo
     
  </p>
</div>
`;
